### PR TITLE
feat(neutron): add ovn support

### PR DESCRIPTION
**Pull request by mnaser**

### CODE REVIEW

**Comment by mnaser** [comment]
@ricolin i addressed your concerns! :)

**Comment by ricolin** [comment]
as disscussed in atmosphere slack, we need to add gateway_ip: null in https://github.com/vexxhost/atmosphere/blob/main/roles/octavia/tasks/generate_resources.yml#L29
With that, I successfully passed deployment and also verify tests

**Comment by mnaser** [comment]
@ricolin final test?  also maybe @okozachenko1203 can also setup a run on their side too.

**Comment by ricolin** [comment]
> @ricolin final test? also maybe @okozachenko1203 can also setup a run on their side too.

as discussed in slack only one test is failing but sounds unrelated
```         "==============================",
        "Failed 1 tests - output below:",
        "==============================",
        "",
        "tempest.api.object_storage.test_container_quotas.ContainerQuotasTest.test_upload_too_many_objects[id-3a387039-697a-44fc-a9c0-935de31f426b,smoke]",
        "------------------------------------------------------------------------------------------------------------------------------------------------",
        "",
        "Captured traceback:",
        "~~~~~~~~~~~~~~~~~~~",
        "    Traceback (most recent call last):",
        "",
        "      File \"/var/lib/openstack/lib/python3.8/site-packages/tempest/common/utils/__init__.py\", line 89, in wrapper",
        "    return func(*func_args, **func_kwargs)",
        "",
        "      File \"/var/lib/openstack/lib/python3.8/site-packages/tempest/api/object_storage/test_container_quotas.py\", line 98, in test_upload_too_many_objects",
        "    self.assertRaises(lib_exc.OverLimit,",
        "",
        "      File \"/var/lib/openstack/lib/python3.8/site-packages/testtools/testcase.py\", line 467, in assertRaises",
        "    self.assertThat(our_callable, matcher)",
        "",
        "      File \"/var/lib/openstack/lib/python3.8/site-packages/testtools/testcase.py\", line 480, in assertThat",
        "    raise mismatch_error",
        "",
        "    testtools.matchers._impl.MismatchError: <bound method ObjectClient.create_object of <tempest.lib.services.object_storage.object_client.ObjectClient object at 0x7fe65057c280>> returned ({'etag': 'd41d8cd98f00b204e9800998ecf8427e', 'last-modified': 'Tue, 25 Jul 2023 18:23:16 GMT', 'x-trans-id': 'tx000005e3c033e730352d2-0064c01313-3ae4-ceph', 'x-openstack-request-id': 'tx000005e3c033e730352d2-0064c01313-3ae4-ceph', 'content-type': 'application/json; charset=utf-8', 'content-length': '0', 'date': 'Tue, 25 Jul 2023 18:23:16 GMT', 'connection': 'close', 'status': '201', 'content-location': 'http://rook-ceph-rgw-ceph.openstack.svc.cluster.local/swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464/OverQuotaObject'}, b'')",
        "",
        "",
        "Captured pythonlogging:",
        "~~~~~~~~~~~~~~~~~~~~~~~",
        "    2023-07-25 18:23:12,394 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:12,882 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"PUT /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464 HTTP/1.1\" 201 0",
        "2023-07-25 18:23:12,884 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:13,701 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"POST /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464 HTTP/1.1\" 204 0",
        "2023-07-25 18:23:13,703 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:14,122 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"PUT /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464/tempest-TestObject-1805920739 HTTP/1.1\" 201 0",
        "2023-07-25 18:23:14,124 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:14,810 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"PUT /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464/tempest-TestObject-893428901 HTTP/1.1\" 201 0",
        "2023-07-25 18:23:14,812 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:15,203 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"PUT /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464/tempest-TestObject-1125896967 HTTP/1.1\" 201 0",
        "2023-07-25 18:23:15,204 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:15,722 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"HEAD /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464 HTTP/1.1\" 204 0",
        "2023-07-25 18:23:15,723 17 DEBUG    [urllib3.connectionpool] Resetting dropped connection: rook-ceph-rgw-ceph.openstack.svc.cluster.local",
        "2023-07-25 18:23:16,104 17 DEBUG    [urllib3.connectionpool] http://rook-ceph-rgw-ceph.openstack.svc.cluster.local:80 \"PUT /swift/v1/889862bd37394e81a9aa7fc858330af6/tempest-TestContainer-153103464/OverQuotaObject HTTP/1.1\" 201 0",
        "",
        "",
        "",
        "======",
        "Totals",
        "======",
        "Ran: 132 tests in 362.8719 sec.",
        " - Passed: 126",
        " - Skipped: 5",
        " - Expected Fail: 0",
        " - Unexpected Success: 0",
        " - Failed: 1",
        "Sum of execute time for each test: 421.9226 sec.",
```

File: roles/octavia/tasks/generate_resources.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@
     network_name: "{{ octavia_management_network_name }}"
     name: "{{ octavia_management_subnet_name }}"
     cidr: "{{ octavia_management_subnet_cidr }}"
+    gateway_ip: null
```
Comment (ricolin) [inline roles/octavia/tasks/generate_resources.yml:29]: Let's remove `gateway_ip` and use `no_gateway_ip` instead
```
no_gateway_ip: true
```

**Comment by ricolin** [comment]
@mnaser 
I just leave comment in https://github.com/vexxhost/atmosphere/pull/499/files#r1274351673

**Comment by mnaser** [comment]
i think this should be full coverage now :)

**Comment by mnaser** [comment]
I just realized we are missing something, we're setting `enable-chassis-as-gw` by default everywhere, when in reality we should only set it on the control plane nodes.  We should find a way to detect the node labels of the system and add that logic.

@okozachenko1203 could you help figure out a smart/effective way to do this?  perhaps the most basic thing is just to have a new DS called ovn-controller-gw and a normal one called ovn-controller?

**Comment by mnaser** [comment]
this failed because of #500 -- ill retrigger it

**Comment by mnaser** [comment]
looks like nova-compute was not up in this case.

**Comment by okozachenko1203** [comment]
debug ci passed without any error lol https://github.com/vexxhost/atmosphere/actions/runs/5717054755/job/15489986606

Let me just rerun

**Pull request merged**